R/writedata.R

```diff
@@ -1,5 +1,92 @@
 #### Output functions
 
+#' Subset a trained emulator.
+#'
+#' A trained fldgen emulator features a large amount of data for both
+#' using the emulator and rigorously validating an emulator.
+#'
+#' If one is just interested in the use of an emulator for generating
+#' felds, this function can be called to reduce a trained emulator to
+#' the bare essential list entries, which can then be saved and called
+#' the same as an unreduced emulator by generate.TP.resids and
+#' generate.TP.fullgrids
+#'
+#' Note that with this reduced emulator, there is NO way to reconstruct
+#' the training data. A fully trained emulator contains a copy of the
+#' training data, in addition to the training regressor values (tgav),
+#' and the estimated linear model parameters and residuals
+#'  (meanfieldT$b, w, r), which together can also reconstruct the data.
+#'
+#' Even though the coordinate information stored in an emulator$griddataT
+#' is not needed directly to generate a new field of residuals or full data,
+#' it is often needed in downstream use of the fields. Therfore an entry
+#' reducedEmulator$griddataT$coord containg a matrix is saved in the
+#' reducedEmulator. Each is a matrix of coordinates for each grid cell, with
+#' cells in rows and latitude, longitude in the two columns. Keeping these
+#' coordinate matrices for T and P is negligible.
+#'
+#' Finally, the reduced emulator produced by this function is specifically
+#' meant for temperature and precipitation only, and is not robust to
+#' extension to other variables.
+#'
+#' Finally finally, if a user is interested in a different subset of
+#' list entries in a trained emulator, they are encouraged to subset and
+#' save themself, as appropriate for their project.
+#'
+#' @param emulator A trained fldgen emulator, with all entries needed
+#' for generating new residuals and for rigourously validating the
+#' quality of the trained emulator
+#'
+#' @return reducedEmulator A trained fldgen emulator with only the list
+#' entries needed by generate.TP.resids and generate.TP.fullgrids for
+#' generating new fields
+#'
+#' @author ACS July 2020
+#' @export
+emulator_reducer <- function(emulator){
+
+    if(length(names(emulator)) < 10){
+        stop('Your emulator is already reduced (missing at least one list entry)')
+    }
+
+    # There _is_ some weirdness to this structure, but it is intentional.
+    # We designed generate.TP.resids and generate.TP.fullgrids to just take
+    # a trained emulator because it was cleaner for the user. That means we need
```

Comment:
You could probably just say that this function reduces the size of the object while preserving the structure expected by generate.TP.resids and generate.TP.fullgrids. 🤷‍♀️